core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java

```diff
@@ -60,9 +41,11 @@ static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCur
     /** The maximum braking force which can be applied at a given speed, in newtons */
     double getMaxBrakingForce(double speed);
 
-    /** The maximum acceleration, in m/s^2, which can be applied at a given speed, in m/s */
-    record TractiveEffortPoint(double speed, double maxEffort) {
-    }
+    /** Returns the different energy sources of the rolling stock */
+    ArrayList<EnergySource> getEnergySources();
+
+    /** If relevant, update the different energy storages states of charge */
+    void updateEnergyStateOfCharge(double force, double speed, double timeStep);
```

Comment:
would find it easier to read with a "Storage" somwhere in the name like :
updateEnergySourceStorageStateOfCharge || updateESStorageStateOfCharge || updateEnergyStorageStateOfCharge